File: HeartForCharity/HeartForCharity.Services/Database/City.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HeartForCharity.Services.Database
{
    public class City
    {
        [Key]
        public int CityId { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = null!;

        [Required]
        [ForeignKey(nameof(Country))]
        public int CountryId { get; set; }

        public virtual Country Country { get; set; } = null!;
        public virtual ICollection<Address> Addresses { get; set; } = new List<Address>();
    }
}

File: HeartForCharity/UI/heartforcharity_shared/lib/model/responses/city.dart
class City {
  final int cityId;
  final String name;
  final int countryId;
  final String countryName;

  City({
    this.cityId = 0,
    this.name = '',
    this.countryId = 0,
    this.countryName = '',
  });

  factory City.fromJson(Map<String, dynamic> json) => City(
        cityId: json['cityId'] ?? 0,
        name: json['name'] ?? '',
        countryId: json['countryId'] ?? 0,
        countryName: json['countryName'] ?? '',
      );
}
